hugr-core/src/hugr.rs

```diff
@@ -213,7 +213,7 @@ impl Hugr {
         //
         // This is not something we want to expose it the API, so we manually
         // iterate instead of writing it as a method.
-        for n in 0..self.node_count() {
+        for n in 0..self.graph.node_capacity() {
```

Comment:
This looks like it belongs in a different PR

Reply:
It's actually a bugfix for the used_extensions resolution. `node_count` doesn't cover all the indices 🤦 

I'll add the drive-by comment

Reply:
Yes please do! The comment above does not explain (to me) why we are using `node_capacity`